vsintegration/src/FSharp.Editor/Completion/CompletionProvider.fs

```diff
@@ -114,7 +114,9 @@ type internal FSharpCompletionProvider
                 getAllSymbols checkFileResults 
                 |> List.filter (fun assemblySymbol ->
                      assemblySymbol.FullName.Contains "." && not (PrettyNaming.IsOperatorDisplayName assemblySymbol.Symbol.DisplayName))
-            let declarations = checkFileResults.GetDeclarationListInfo(Some(parseResults), fcsCaretLineNumber, line, partialName, getAllSymbols)
+            let completionContextPos = Position.fromZ caretLinePos.Line caretLinePos.Character
+            let completionContext = ParsedInput.TryGetCompletionContext(completionContextPos, parseResults.ParseTree, line)
+            let declarations = checkFileResults.GetDeclarationListInfo(Some(parseResults), fcsCaretLineNumber, line, partialName, Some(completionContextPos, completionContext), getAllSymbols)
```

Comment:
Can we pass in the completionContextPos to GetDeclarationListInfo and internalize the call to TryGetCompletionContext?